src/ioctls/vcpu.rs

```diff
@@ -2172,4 +2205,35 @@ mod tests {
             vcpu.enable_cap(&cap).unwrap();
         }
     }
+    #[cfg(target_arch = "x86_64")]
+    #[test]
+    fn test_get_tsc_khz() {
+        let kvm = Kvm::new().unwrap();
+        // Skip test if GetTscKhz is missing.
+        if !kvm.check_extension(Cap::GetTscKhz) {
+            return;
+        }
+
+        let vm = kvm.create_vm().unwrap();
+        let vcpu = vm.create_vcpu(0).unwrap();
+        assert!(vcpu.get_tsc_khz().unwrap() > 0);
+    }
+
+    #[cfg(target_arch = "x86_64")]
+    #[test]
+    fn test_set_tsc_khz() {
+        let kvm = Kvm::new().unwrap();
+        // Skip test if KVM/host capabilities are missing.
+        if !(kvm.check_extension(Cap::GetTscKhz) && kvm.check_extension(Cap::TscControl)) {
```

Comment:
For the hosts that are missing these capabilities we could test that `set/get` return an error.

Reply:
Done.